Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -3239,25 +3239,39 @@ static RefPtr<CSSPrimitiveValue> consumeBackgroundSize(CSSPropertyID property, C
     if (identMatches<CSSValueContain, CSSValueCover>(range.peek().id()))
         return consumeIdent(range);
 
+    auto identicalValueEncoding = Pair::IdenticalValueEncoding::DoNotCoalesce;
+
     // FIXME: We're allowing the unitless quirk on this property because our
     // tests assume that. Other browser engines don't allow it though.
     RefPtr<CSSPrimitiveValue> horizontal = consumeIdent<CSSValueAuto>(range);
-    if (!horizontal)
+    if (horizontal)
+        identicalValueEncoding = Pair::IdenticalValueEncoding::Coalesce;
+    else
         horizontal = consumeLengthOrPercent(range, cssParserMode, ValueRange::NonNegative, UnitlessQuirk::Allow);
 
+    if (!horizontal)
+        return nullptr;
+
     RefPtr<CSSPrimitiveValue> vertical;
     if (!range.atEnd()) {
-        if (range.peek().id() == CSSValueAuto) // `auto' is the default
-            range.consumeIncludingWhitespace();
-        else
+        vertical = consumeIdent<CSSValueAuto>(range);
+        if (!vertical)
             vertical = consumeLengthOrPercent(range, cssParserMode, ValueRange::NonNegative, UnitlessQuirk::Allow);
-    } else if (!vertical && property == CSSPropertyWebkitBackgroundSize) {
-        // Legacy syntax: "-webkit-background-size: 10px" is equivalent to "background-size: 10px 10px".
-        vertical = horizontal;
+        if (range.atEnd() && !vertical)
```

Comment:
Can this happen? If `!vertical` it means that both `consumeIdent` and `consumeLengthOrPercent` failed, so nothing should have been consumed and `!range.atEnd()` should still hold?

Reply:
Good catch, I have removed this logic.